### PR TITLE
Subtests for issue #1520

### PR DESCRIPTION
**Pull request by petdance**
### Summary
Convert three .t files to use subtest in response to issue #1520.

### CODE REVIEW

File: t/mojo/user_agent_socks.t [hunk 1]
```diff
@@ -99,67 +99,78 @@ Mojo::IOLoop->singleton->reactor->io(
...
+  ok $tx->kept_alive, 'kept connection alive';
+  ok $tx->keep_alive, 'keep connection alive';
+  is $tx->res->code, 200, 'right status';
+  is $tx->res->body, $last, 'right content';
+  is $before, $last, 'same port';
+  isnt(Mojo::IOLoop->stream($tx->connection)->handle->sockport, $last, 'different ports');
+};
+
+subtest 'WebSocket with SOCKS proxy' => sub {
+  my $ua = Mojo::UserAgent->new(ioloop => Mojo::IOLoop->singleton, insecure => 1);
```
Comment (kraih) [inline t/mojo/user_agent_socks.t:136]: This test doesn't appear to have a SOCKS proxy configured in the new user agent.

File: t/mojo/response.t [hunk 2]
```diff
@@ -9,218 +9,231 @@ use Mojo::JSON qw(encode_json);
 use Mojo::Message::Response;
 use Mojo::Util qw(encode gzip);
 
-# Defaults
-my $res = Mojo::Message::Response->new;
-is $res->max_message_size, 2147483648, 'right default';
+# $dir is used in multiple subtests.
+my $dir = tempdir;
```
Comment (kraih) [inline t/mojo/response.t:13]: It's only used in two places, that's not really worth sharing. The two tests can just have their own tempdir each.

File: t/mojo/user_agent_socks.t [hunk 3]
```diff
@@ -99,67 +99,78 @@ Mojo::IOLoop->singleton->reactor->io(
...
+  ok !$tx->kept_alive, 'kept connection not alive';
+  ok $tx->keep_alive, 'keep connection alive';
+  is $tx->res->code, 200, 'right status';
+  is $tx->req->headers->proxy_authorization, undef, 'no "Proxy-Authorization" value';
+  is $tx->res->body, $last, 'right content';
+  isnt(Mojo::IOLoop->stream($tx->connection)->handle->sockport, $last, 'different ports');
+};
+
+subtest 'Keep alive request with SOCKS proxy' => sub {
+  my $ua = Mojo::UserAgent->new(ioloop => Mojo::IOLoop->singleton, insecure => 1);
```
Comment (kraih) [inline t/mojo/user_agent_socks.t:123]: This test doesn't appear to have a SOCKS proxy configured in the new user agent.

File: t/mojo/response.t (hunk 2 above)
Comment (petdance) [reply t/mojo/response.t:13]: That's fine with me. I didn't want to potentially cause any changes in behavior by using two different temp dirs.

File: t/mojo/response.t [hunk 4]
```diff
@@ -9,218 +9,231 @@ use Mojo::JSON qw(encode_json);
 use Mojo::Message::Response;
 use Mojo::Util qw(encode gzip);
 
-# Defaults
-my $res = Mojo::Message::Response->new;
-is $res->max_message_size, 2147483648, 'right default';
+# $dir is used in multiple subtests.
```
[the remark this answers is not in the document]
Comment (petdance) [reply t/mojo/response.t:12]: Removed since that $dir is going away.

How should that comment have been done?

**Comment by kraih** [comment]
Please squash perltidy commits into the relevant "feature" commits.

**Comment by petdance** [comment]
Do you want one commit for each of the test files? Or should all three go into one commit?

**Comment by kraih** [comment]
One commit for everything would be fine too. The preferred version would be one commit per test file though.

**Comment by petdance** [comment]
Then that's what I'll do.

**Comment by petdance** [comment]
I've redone this PR as #1831

**Pull request closed without merging**